### DISCUSSION TITLE
Difference between plain and non plain functions?

### DISCUSSION THREAD

**Discussion by Madoxen** [go-git/go-git#1110] [General]
There are functions that begin with `Plain*` and all of these have a non-plain equivalent. What is the difference between those two families of functions?

**Comment by pjbgf** [comment]
@Madoxen `Plain` generally have fewer options and will always Clone/Open from the actual disk. The non-plain versions allows you to set the `Filesystem` to be used, which can be anything supported by go-billy's [Filesystem interface](https://github.com/go-git/go-billy/blob/066ee277a284d1b95037d8f7d3d5cd535b9aab84/fs.go#L52-L58).

**Reply by Madoxen** [reply]
Thanks, that clears things up!
